test/directive_test.js

```diff
@@ -0,0 +1,269 @@
+'use strict';
+
+const angular = require('angular');
+require('angular-mocks');
+require('../app/js/client');
+
+const dummyTemplate = require('../app/templates/first/dummy.html');
+const fullImageTemplate = require('../app/templates/first/full_image_template.html');
+const tinyImageTemplate = require('../app/templates/first/tiny_image_template.html');
+const titleImageTemplate = require('../app/templates/first/title_image_template.html');
+const albumsTemplate = require('../app/templates/first/albums.html');
+
+describe('directive tests', () => {
+  let $httpBackend;
+  let $scope;
+  let $compile;
+
+  beforeEach(() => {
+    angular.mock.module('ButtSnifferApp');
+    angular.mock.inject(function(_$httpBackend_, $rootScope,
+    _$compile_) {
+      $scope = $rootScope.$new();
+      $compile = _$compile_;
+      $httpBackend = _$httpBackend_;
+    });
+  });
+
+  it('should be a test', () => {
+    expect(true).toBe(true);
+  });
+
+  it('should be a dummy', () => {
+    $httpBackend.expectGET('./templates/first/dummy.html')
+      .respond(200, dummyTemplate);
+    $scope.test = 'test data';
+    let link = $compile('<dummy data="test"></dummy>');
+    let directive = link($scope);
+    $scope.$digest();
+    $httpBackend.flush();
+
+    let h3 = directive.find('h3');
+    let text = h3.text();
+
+    expect(text).toBe('test data');
+  });
+
+  it('full image title property should work', () => {
+    $httpBackend.expectGET('./templates/first/full_image_template.html')
+      .respond(200, fullImageTemplate);
+    $scope.test = 'test title';
+    let link = $compile('<full-image-directive title="test"></full-image-directive>');
+    let directive = link($scope);
+    $scope.$digest();
+    $httpBackend.flush();
+
+    let h3 = directive.find('h3');
+    let text = h3.text();
```

Comment:
You might run into a problem here when you are finding 'h3' since you have more than one 'h3' in your template.  Just be aware of that in case you end up with unexpected results.  Same thing in the other tests where you are doing this similar thing.